server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -126,10 +126,11 @@ protected void asyncShardOperation(GetRequest request, ShardId shardId, ActionLi
         IndexService indexService = indicesService.indexServiceSafe(shardId.getIndex());
         IndexShard indexShard = indexService.getShard(shardId.id());
         if (indexShard.routingEntry().isPromotableToPrimary() == false) {
+            assert indexShard.indexSettings().isFastRefresh() == false
```

Comment:
I have no idea how serverless upgrade tests are working and we'll see what CI says, but I suppose this can trip if a request is wrongly redirected in a mixed cluster.

Reply:
You're right, lets not risk it. I'll comment the assertion and add a TODO.